alf/utils/distributed.py

```diff
@@ -62,11 +64,22 @@ def __init__(self, module: torch.nn.Module, perform: Callable[..., Any]):
         # We also need to ignore all the buffers that is under the replay buffer
         # of the module (e.g. when the module is an Algorithm) for DDP, because
         # we do not want DDP to synchronize replay buffers across processes.
+        #
         # Those buffers are not registered in the state_dict() because of Alf's
         # special treatment but can be found under named_buffers(). We do not
         # want DDP to synchronize replay buffers.
-        for name, _ in self.named_buffers():
-            if '_replay_buffer' in name.split('.'):
+        ignored_named_buffers = []
+        for sub_module in module.modules():
+            if isinstance(sub_module, ReplayBuffer):
+                for _, buf in sub_module.named_buffers():
+                    # Find all the buffers that are registered under a
+                    # ReplayBuffer submodule.
+                    ignored_named_buffers.append(buf)
+
+        for name, buf in self.named_buffers():
+            # If the buffer is in the ignored_named_buffers (address-wise equal,
+            # i.e. ``is``), add its name to DDP's ignore list.
+            if any([buf is x for x in ignored_named_buffers]):
```

Comment:
you can make ignore_named_buffers a set and here will be `if x in ignored_named_buffers`

Reply:
Agree and that was how I originally implement this, then I realized `in set` does not explicitly check for address equality, and I would like make sure to use `is` for address equality here.

Reply:
I think there might be ways to make `set` use address (a.k.a `id`) to check for equality, but that would be at the cost of readability ... so I compromised and chose to use this.

Reply:
Turns out I was over-thinking. I verified that `set` actually test membership in the desired way. Updated the PR as suggested.